modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/GlobalSnapshotStateChannelEventsProcessor.scala

```diff
@@ -114,74 +128,125 @@ object GlobalSnapshotStateChannelEventsProcessor {
           .map(_.snapshotInfo)
 
       def processCurrencySnapshots(
+        snapshotOrdinal: SnapshotOrdinal,
         lastGlobalSnapshotInfo: GlobalSnapshotInfo,
         events: SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]]
-      )(implicit hasher: Hasher[F]): F[SortedMap[Address, NonEmptyList[CurrencySnapshotWithState]]] =
-        events.toList
-          .foldLeftM(SortedMap.empty[Address, NonEmptyList[CurrencySnapshotWithState]]) {
-            case (agg, (address, binaries)) =>
-              type Success = NonEmptyList[CurrencySnapshotWithState]
-              type Result = Option[Success]
-              type Agg = (Result, List[Signed[StateChannelSnapshotBinary]])
-
-              val initialState = lastGlobalSnapshotInfo.lastCurrencySnapshots.get(address)
-
-              (NonEmptyList.fromList(initialState.toList), binaries.toList.reverse)
-                .tailRecM[F, Result] {
-                  case (state, Nil) => state.asRight[Agg].pure[F]
-
-                  case (None, head :: tail) =>
-                    jsonBrotliBinarySerializer
-                      .deserialize[Signed[CurrencySnapshot]](head.value.content)
-                      .map {
-                        _.toOption.map { snapshot =>
-                          (NonEmptyList.one(snapshot.asLeft).some, tail)
-                            .asLeft[Result]
-                        }
+      )(implicit hasher: Hasher[F]): F[SortedMap[Address, MetagraphAcceptanceResult]] = {
+        val isFeeRequired = feeCalculator.isFeeRequired(snapshotOrdinal)
+
+        def deserialize[A: Decoder](binary: Signed[StateChannelSnapshotBinary]): F[Option[A]] =
+          jsonBrotliBinarySerializer.deserialize[A](binary.value.content).map(_.toOption)
```

Comment:
I see doing `toOption` is the same as before but do you know why we are throwing away the error?

Reply:
Because we don't know if the deserialization is going to work - it may or may not be a `CurrencySnapshot`/`CurrencyIncrementalSnapshot`

Reply:
So deserializing error is not unexpected in any way.